Tella/Data/Crypto/Cryptor.swift

```diff
@@ -0,0 +1,161 @@
+        
+        // Open the input and output file handles
+        self.inputFile = try FileHandle(forReadingFrom: inputFileURL)
+        self.outputFile = try FileHandle(forWritingTo: outputFileURL)
+        
+        self.cryptoOperation = cryptoOperation
+        self.encryptionKeyData = encryptionKeyData
+    }
+    
+    func cryptFile() throws {
+        try cryptorCreate()
+        try cryptorUpdate()
+        try cryptorFinal()
+    }
+    
+    private func cryptorCreate() throws  {
+        // Setup the encryption context
+        let keySize = size_t(kCCKeySizeAES256)
+        let algorithm = CCAlgorithm(kCCAlgorithmAES)
+        let options = CCOptions(ccNoPadding)
+        let mode = CCMode(kCCModeCTR)
+        let operation: CCOperation = cryptoOperation == .encrypt ? UInt32(kCCEncrypt) : UInt32(kCCDecrypt)
+        
+        let operationResult = encryptionKeyData.withUnsafeBytes { keyBytes in
+            
+            CCCryptorCreateWithMode(
+                operation,
+                algorithm,
+                mode,
+                options,
+                nil, // Initialization Vector
+                keyBytes.baseAddress, // Encryption Key
+                keySize,
+                nil,
+                0,
+                0,
+                CCModeOptions(kCCModeOptionCTR_BE),
+                &cryptor
+            )
+        }
+        
+        if operationResult !=  kCCSuccess {
+            debugLog("Error encryption \(operationResult)", space: .crypto)
+            throw CryptoError.cryptorCreationFailed
+        }
+    }
+    
+    private func cryptorUpdate() throws {
+        var stop = false
+        
+        // Process the file in chunks
+        while true {
+            
+            try autoreleasepool {
+                
+                // Read a chunk from the input file
+                
+                let inputBuffer = inputFile.readData(ofLength: Self.bufferSize) as Data
+                
+                var encryptedData = Data(count: Int(Self.bufferSize) + kCCBlockSizeAES128)
+                
+                if inputBuffer.isEmpty {
+                    // Reached end of file
+                    stop = true
+                }
+                
+                // Update the encryption context with the input chunk and write the output to the output file
+                var encryptedByteCount = 0
+                
+                let operationResult = inputBuffer.withUnsafeBytes { inputBytes in
+                    encryptedData.withUnsafeMutableBytes { outputBytes in
+                        CCCryptorUpdate(
+                            cryptor!,
+                            inputBytes.baseAddress, inputBytes.count,
+                            outputBytes.baseAddress, outputBytes.count,
+                            &encryptedByteCount
+                        )
+                    }
+                }
+                
+                if operationResult != kCCSuccess {
```

Comment:
if we add a defer we could get rid of the closeFile statements here and change this into a guard statement

```
guard operationResult == kCCSuccess else {
  throw CryptoError.encryptionFailed
}
```